src/m365/aad/commands/user/user-set.ts

```diff
@@ -197,9 +337,27 @@ class AadUserSetCommand extends GraphCommand {
     if (options.resetPassword) {
       requestBody.passwordProfile = {
         forceChangePasswordNextSignIn: options.forceChangePasswordNextSignIn || false,
+        forceChangePasswordNextSignInWithMfa: options.forceChangePasswordNextSignInWithMfa || false,
         password: options.newPassword
       };
     }
+    const propertyMap: any = {
+      displayName: options.displayName,
+      givenName: options.firstName,
+      surname: options.lastName,
+      usageLocation: options.usageLocation,
+      officeLocation: options.officeLocation,
+      jobTitle: options.jobTitle,
+      companyName: options.companyName,
+      department: options.department,
+      preferredLanguage: options.preferredLanguage
+    };
+
+    Object.keys(propertyMap).forEach(key => {
+      if (propertyMap[key]) {
+        requestBody[propertyMap[key]] = propertyMap[key];
+      }
+    });
```

Comment:
I wrote this because otherwise it would make the tests `correctly changes password for current user retrieved by userPrincipalName` and `correctly changes password for current user retrieved by objectId` fail. Since the manifest object contains all undefined properties, it makes the post request where it updates the user properties, which it shouldn't. see line 257 in `user-set.ts`

Reply:
That's unfortunate indeed, but nothing we can't fix. We could replace that check with something like
```typescript
if (Object.keys(body).some(k => body[k] !== undefined)) {
// ... execute request
}

Reply:
That's indeed the case, thank you for helping me here!